src/export/InstanceExporter.ts

```diff
@@ -91,17 +91,21 @@ export class InstanceExporter implements Fishable {
           const foundFixedValue = cloneDeep(associatedEl[fixedValueKey as keyof ElementDefinition]);
           if (foundFixedValue) {
             // Find how much the two paths overlap, for example, a.b.c, and a.b.d overlap for a.b
-            let overlapIndex = 0;
-            element.id.split('.').forEach((p, i) => {
-              if (p === associatedEl.id.split('.')[i]) overlapIndex++;
-            });
+            let overlapIdx = 0;
+            const elParts = element.id.split('.');
+            const assocElParts = associatedEl.id.split('.');
+            for (
+              ;
+              overlapIdx < elParts.length && elParts[overlapIdx] === assocElParts[overlapIdx];
+              overlapIdx++
+            );
             // We must keep the relevant portion of the beginning of path to preserve sliceNames
             // and combine this with portion of the associatedEl's path that is not overlapped
-            const pathStart = splitOnPathPeriods(path).slice(0, overlapIndex - 1);
+            const pathStart = splitOnPathPeriods(path).slice(0, overlapIdx - 1);
             const pathEnd = associatedEl
               .diffId()
               .split('.')
-              .slice(overlapIndex)
+              .slice(overlapIdx)
```

Comment:
I know you just updated this part to be cleaner, but: you could also consider using [takeWhile](https://lodash.com/docs/4.17.15#takeWhile) from lodash.

Reply:
Ah cool, that looks like the function that I felt like should've existed but I could not find, thanks! In this case I don't know how I would go about using it, because I find the overlap from the **FHIR** paths on the ElementDefinitions of the elements, and then use that overlap to slice up the corresponding **FSH** versions of the paths. So I'm not sure how I'd simplify things with `takeWhile` here.

Reply:
Ah, that's a good point. It might be more troublesome than useful. I think it would look something like this:
```
const pathStart = _.takeWhile(splitOnPathPeriods(path), (val, ind) => _.nth(elParts, ind) == _.nth(assocElParts, ind));
const pathEnd = associatedEl.diffId().split('.').slice(pathStart.length()).map(...);
```

Also, looking at the indices, shouldn't the slice for `pathEnd` start at `overlapIdx - 1`? I know different arrays are being sliced, but this doesn't use the path part at index `overlapIdx - 1` from either path array.

Reply:
The `pathStart` is taken from `path`, which is a pure FSH path, so it doesn't include the extra resourceType at the beginning of the path (so just `a.b.c`, not `Patient.a.b.c`). The `pathEnd` is taken from a FHIR id that id getting converted into a FSH path, so it does have the resourceType appended to the beginning of the path (`Patient.a.b.c`, not `a.b.c`). So let's say the original path was `a.b.c`, and we get element `Patient.a.b.c` from this. Then the associated element path could be `Patient.a.b.d`, so the overlap is 3. From the original path (`a.b.c`), we want `a.b`, and from the associated element path, we want `d`. So to get `a.b` from `a.b.c`, we need to subtract 1 from the overlap index, since there is no `Patient` at the beginning of the `path`.